### PR TITLE
Add install capabilities to top-level CMake config

### PR DESCRIPTION
**Pull request by constantfold**
Before this change, the generated Makefile by CMake lacked the capability to run `make install`. This PR makes it possible
to run `make install`, for example:

```bash
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=RELEASE -DCASS_INSTALL_PKG_CONFIG=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib64 ..
make DESTDIR=/tmp/example install
```

The names and contents of installed files try to mimic the original C++ Driver files. For example, installation of C++ Driver generates the following `.so` files/symlinks:

```
usr/lib64/libscylla-cpp-driver.so -> libscylla-cpp-driver.so.2
usr/lib64/libscylla-cpp-driver.so.2 -> libscylla-cpp-driver.so.2.16.2
usr/lib64/libscylla-cpp-driver.so.2.16.2
```

This particular behavior is replicated in a new `build-cmake-compat.sh` script, which is executed as a `BUILD_COMMAND` in the Rust external project in the top-level `CMakeLists.txt`.

Unfortunately, I have not been able to configure CMake to create such symlinks on its own. One of the reasons is an unfixed bug in CMake: https://gitlab.kitware.com/cmake/cmake/-/issues/23249#note_1277271. Therefore, the `build-cmake-compat.sh` is introduced to create the necessary files in a way it's easy to install for CMake.

The third commit includes the static library in the installed files. 

Those are the contents of `DESTDIR` (as in `make DESTDIR=... install`) in the original C++ driver:
```
DESTDIR
└── usr
    ├── include
    │   ├── cassandra.h
    │   └── dse.h
    └── lib64
        ├── libscylla-cpp-driver.so -> libscylla-cpp-driver.so.2
        ├── libscylla-cpp-driver.so.2 -> libscylla-cpp-driver.so.2.16.2
        ├── libscylla-cpp-driver.so.2.16.2
        └── libscylla-cpp-driver_static.a
```

and after this PR in the bindings:
```
DESTDIR
└── usr
    ├── include
    │   └── cassandra.h
    └── lib64
        ├── libscylla-cpp-driver.so -> libscylla-cpp-driver.so.0
        ├── libscylla-cpp-driver.so.0 -> libscylla-cpp-driver.so.0.0.1
        ├── libscylla-cpp-driver.so.0.0.1
        └── libscylla-cpp-driver_static.a
```

This PR is a prerequisite for a follow-up PR reworking #88.

## Pre-review checklist

- [X] I have split my patch into logically separate commits.
- [X] All commit messages clearly explain what they change and why.
- [X] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have enabled appropriate tests in `.github/workflows/build.yml` in `gtest_filter`.
- [ ] I have enabled appropriate tests in `.github/workflows/cassandra.yml` in `gtest_filter`.

### CODE REVIEW

**Review by Gor027 — COMMENTED** [review]
I manually enabled the flag `CASS_BUILD_STATIC` in `CMakeLists.txt`, however, after the build finished `.a` file was not created in the destination directory. I built it with the same commands as in the PR description. Should I build it in another way?

File: scylla-rust-wrapper/build-cmake-compat.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -e
+
+# The version number should be in the third line
+# of Cargo.toml file (e.g. version = "0.0.1")
+VERSION_NUMBER=$(sed -n '3p' < Cargo.toml)
```
Comment (wprzytula) [inline scylla-rust-wrapper/build-cmake-compat.sh:6]: Isn't this an assumption that is prone to change?

File: scylla-rust-wrapper/build-cmake-compat.sh [hunk 2]
```diff
@@ -17,11 +17,31 @@ else
...
+fi
+
+
 create_symlinks() {
   rm -rf target/$1/cmake-compat/
   mkdir target/$1/cmake-compat/
 
   cp target/$1/libscylla_cpp_driver.so target/$1/cmake-compat/libscylla-cpp-driver.so.$MAJOR.$MINOR.$PATCH
+  
+  if [[ "$COPY_STATIC" == true ]]
```
Comment (wprzytula) [inline scylla-rust-wrapper/build-cmake-compat.sh:40]: Is comparison to `true` necessary in Bash??

**Review by wprzytula — APPROVED** [review]
I've tried it out and the results were very good. The `-DCASS_BUILD_STATIC` option worked as well, respecting correctly options: `=false`, `=OFF` as non-static build and `=true`, `=ON` as static. The `.a` file was present in the target dir.

**Comment by roydahan** [comment]
@avelanarius is this one still needed?
If so, what's preventing it from being merged?

**Comment by roydahan** [comment]
Closing in favor of #88

**Comment by Lorak-mmk** [comment]
Did you mean #122 ?

**Comment by roydahan** [comment]
Yes

**Pull request closed without merging**